traits/src/collection.rs

```diff
@@ -0,0 +1,53 @@
+use codec::{Decode, Encode};
+use scale_info::TypeInfo;
+use sp_runtime::{DispatchError, DispatchResult, RuntimeDebug};
+use sp_std::{cmp::Eq, result};
+
+/// Collection info.
+#[cfg_attr(feature = "std", derive(PartialEq, Eq))]
+#[derive(Encode, Decode, RuntimeDebug, TypeInfo)]
+pub struct CollectionInfo<BoundedString, AccountId> {
+	/// Current bidder and bid price.
+	pub issuer: AccountId,
+	pub metadata: BoundedString,
+	pub max: u32,
+	pub symbol: BoundedString,
+	pub id: BoundedString,
+}
+
+/// Abstraction over a Collection system.
+#[allow(clippy::upper_case_acronyms)]
+pub trait Collection<BoundedString, AccountId> {
+	type CollectionId: Default + Copy;
+	// fn collection_two_info(
+	// 	id: Self::CollectionTwoId,
+	// ) -> Option<CollectionTwoInfo<AccountId, BoundedString>>;
+	fn issuer(collection_id: Self::CollectionId) -> Option<AccountId>;
+	fn create_collection(
+		issuer: AccountId,
+		metadata: BoundedString,
```

Comment:
Does the metadata has he same max length as the symbol?

Reply:
It could be smaller. Currently StringLimit for metadata is set to u128. Symbol could be something like u32